### PR TITLE
IPC-conf WIP dbus

### PR DESCRIPTION
**Pull request by omarhogni**

### CODE REVIEW

File: libs/ipc_connector/tests/ipc_connector_test.cpp [hunk 1]
```diff
@@ -15,39 +15,24 @@ auto main(int argc, char** argv) -> int {
...
-    });
-
-    my_signal->async_send("helloworld", [](auto const&, auto) {});
-
-    ctx.run_for(std::chrono::milliseconds(100));
-
-    expect(called);
-  };
-
+  // "ipc_connector test"_test = []() {
```
Comment (jbbjarnason) [inline libs/ipc_connector/tests/ipc_connector_test.cpp]: maybe move to example if this does not test

File: libs/ipc_connector/tests/ipc_connector_test.cpp [hunk 2]
```diff
@@ -15,39 +15,24 @@ auto main(int argc, char** argv) -> int {
...
+
+  boost::asio::io_context ctx;
+  tfc::ipc_ruler::ipc_manager_client client(ctx);
+
+  client.register_signal("signal", tfc::ipc::type_e::_bool, [](const boost::system::error_code& ec) {
+    if (ec) {
+      std::cerr << "Error occured" << ec.what() << std::endl;
+    }
+  });
+  client.register_slot(fmt::format("slot"), tfc::ipc::type_e::_bool, [](const boost::system::error_code& ec) {
```
Comment (jbbjarnason) [inline libs/ipc_connector/tests/ipc_connector_test.cpp:26]: unnecessary fmt::format

File: libs/ipc_connector/tests/ipc_connector_test.cpp [hunk 3]
```diff
@@ -15,39 +15,24 @@ auto main(int argc, char** argv) -> int {
...
+    if (ec) {
+      std::cerr << "Error occured" << ec.what() << std::endl;
+    }
+  });
+
+  std::cout << "HERE" << std::endl;
```
Comment (jbbjarnason) [inline libs/ipc_connector/tests/ipc_connector_test.cpp]: can be removed

**Review by jbbjarnason — APPROVED** [review]

File: libs/dbus_utils/tests/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,15 @@
+
+find_package(ut CONFIG REQUIRED)
```
[the remark this answers is not in the document]
Comment (omarhogni) [reply libs/dbus_utils/tests/CMakeLists.txt:2]: bætti við smá testi til að halda folder structure eins.

File: libs/ec/inc/public/tfc/ec/devices/abt/easycat.hpp [hunk 5]
```diff
@@ -14,25 +14,17 @@ class easyecat final : public base {
...
-          tfc::ipc::bool_send::create(ctx_, fmt::format("easyecat.{}.bool.in.{}", slave_index, i));
-      if (!ptr) {
-        printf("%s\n", ptr.error().message().c_str());
-        std::terminate();
-      }
-      bool_transmitters_.push_back(ptr.value());
+      // std::unique_ptr<tfc::ipc::bool_send_exposed> ptr =
+      //     std::make_unique<tfc::ipc::bool_send_exposed>();
+      bool_transmitters_.emplace_back(
+          std::make_unique<tfc::ipc::bool_send_exposed>(ctx_, fmt::format("easyecat.{}.bool.in.{}", slave_index, i)));
```
[the remark this answers is not in the document]
Comment (omarhogni) [reply libs/ec/inc/public/tfc/ec/devices/abt/easycat.hpp:20]: no.

File: libs/ipc/inc/public/tfc/ipc.hpp [hunk 6]
```diff
@@ -55,13 +55,12 @@ class transmission_base {
...
 
-  [[nodiscard]] auto endpoint() const -> std::string {
-    return utils::socket::zmq::ipc_endpoint_str(
-        fmt::format("{}.{}.{}", base::get_exe_name(), base::get_proc_name(), name_w_type()));
-  }
+  [[nodiscard]] auto endpoint() const -> std::string { return utils::socket::zmq::ipc_endpoint_str(name_w_type()); }
 
   /// \return <type>.<name>
-  [[nodiscard]] auto name_w_type() const -> std::string { return fmt::format("{}.{}", type_desc::type_name, name_); }
+  [[nodiscard]] auto name_w_type() const -> std::string {
```
[the remark this answers is not in the document]
Comment (omarhogni) [reply libs/ipc/inc/public/tfc/ipc.hpp:61]: exe and id are part of the name. name_w_type should have been partial_name_w_type()

name_w_type was also always used with the exe and id added to it.

File: libs/ipc_connector/inc/public/tfc/ipc_connector.hpp [hunk 7]
```diff
@@ -18,22 +17,22 @@ class slot_configurable {
...
-            } else {
-              slot_->init(new_signal_name, callback);
-            }
-          });
-        }) {}
+      : slot_(slot_callback<type_desc>::create(ctx, name)), client_(ctx) {
+    client_.register_connection_change_callback([this, callback](std::string slot_name, std::string signal_name) {
+      if (slot_name == slot_->name_w_type()) {
+        if (signal_name.empty()) {
+          slot_->disconnect("");
```
[the remark this answers is not in the document]
Comment (omarhogni) [reply libs/ipc_connector/inc/public/tfc/ipc_connector.hpp:24]: I need to test this.

File: libs/ipc_connector/inc/public/tfc/ipc_connector/dbus_server_iface.hpp [hunk 8]
```diff
@@ -0,0 +1,255 @@
...
+    std::string last_registered;
+    std::string last_modified;
+    std::string modified_by;
+    std::string connected_to;
+  };
+
+  explicit ipc_manager(std::function<void(std::string, std::string)> on_connect_cb)
+      : logger_("ipc_manager"), on_connect_cb_{ std::move(on_connect_cb) } {}
+
+  auto register_signal(std::string name, tfc::ipc::type_e type) -> void {
```
[the remark this answers is not in the document]
Comment (omarhogni) [reply libs/ipc_connector/inc/public/tfc/ipc_connector/dbus_server_iface.hpp:55]: No not any in particular. The callback I passed in below took in a const& std::string but i switched it out for std::string_view

File: libs/ipc_connector/inc/public/tfc/ipc_connector/dbus_server_iface.hpp [hunk 9]
```diff
@@ -0,0 +1,255 @@
...
+        std::tuple<std::string, uint8_t, std::string, std::string, std::string, std::string, std::string, std::string>>
+        ret;
+    for (auto& [_, slot] : slots_) {
+      ret.emplace_back(slot.name, static_cast<uint8_t>(slot.type), slot.created_by, slot.created_at, slot.last_registered,
+                       slot.last_modified, slot.modified_by, slot.connected_to);
+    }
+    return ret;
+  }
+
+  auto connect(const std::string& slot_name, const std::string& signal_name) -> void {
```
[the remark this answers is not in the document]
Comment (omarhogni) [reply libs/ipc_connector/inc/public/tfc/ipc_connector/dbus_server_iface.hpp]: It does not need to disconnect first.

File: libs/ipc_connector/inc/public/tfc/ipc_connector.hpp (hunk 7 above)
[the remark this answers is not in the document]
Comment (omarhogni) [reply libs/ipc_connector/inc/public/tfc/ipc_connector.hpp:24]: This disconnect does not work.

Initalizing the slot with a new signal of an empty name Does how ever work.
```cpp
      if (slot_name == slot_->name_w_type()) {
          slot_->init(signal_name, callback);
      }
``` 

skipping the disconnect thought all together.

**Review by jbbjarnason — COMMENTED** [review]
foo

**Pull request merged**